tests/test_modelgen.py

```diff
@@ -314,28 +328,59 @@ def test_InceptionTime_hyperparameters(self):
 
-        hyperparams = modelgen.generate_InceptionTime_hyperparameter_set(custom_settings)
+        model_type = Model_InceptionTime(x_shape, None, **custom_settings)
+        hyperparams = model_type.generate_hyperparameters()
 
         assert hyperparams.get('network_depth') == 5, 'Wrong network depth'
         assert hyperparams.get('max_kernel_size') == 10, 'Wrong kernel'
         assert hyperparams.get('filters_number') == 32, 'Wrong filter number'
 
     # Tests for general mcfly functionality:
     def test_generate_models_metrics(self):
-        """ "Test if correct number of models is generated and if metrics is correct. """
+        """ Test if correct number of models is generated and if metrics is correct. """
         x_shape = (None, 20, 10)
         nr_classes = 2
         X_train, y_train = self._generate_train_data(x_shape, nr_classes)
+        n_models = 5
 
-        models = modelgen.generate_models(x_shape, nr_classes, 5)
+        models = modelgen.generate_models(x_shape, nr_classes, n_models)
         for model in models:
-            model[0].fit(X_train, y_train)
+            model[0].fit(X_train, y_train, epochs = 1)
 
         model, hyperparams, modeltype = models[0]
         model_metrics = [m.name for m in model.metrics]
-        assert "accuracy" in model_metrics
-        assert len(models) == 5
+        assert "accuracy" in model_metrics, "Not found accuracy for model {}. Found {}".format(
+            modeltype, model_metrics)
+        assert len(models) == n_models, "Expecting {} models, found {} models".format(
+            n_models, len(models))
+
+    def test_generate_models_pass_model_object(self):
+        """ Test if model class can be passed as model_types input."""
+        x_shape = (None, 20, 10)
+        nr_classes = 2
+        X_train, y_train = self._generate_train_data(x_shape, nr_classes)
+        n_models = 4
+        
+        models = modelgen.generate_models(x_shape, nr_classes, n_models,
+                                          model_types=['CNN', Model_ResNet])
+        created_model_names = list(set([x[2] for x in models]))
+        created_model_names.sort()
+        assert len(models) == 4, "Expected number of models to be 4"
+        assert created_model_names == ["CNN", "ResNet"], "Expected different model names."
+        for model in models:
+            assert isinstance(model[0], Model), "Expected keras model."
 
+    def test_generate_models_exception(self):
+        """ Test expected generate_models exception."""
+        x_shape = (None, 20, 10)
+        nr_classes = 2
+        X_train, y_train = self._generate_train_data(x_shape, nr_classes)
+        n_models = 2
+        
+        with pytest.raises(NameError, match="Unknown model name given."):
+            models = modelgen.generate_models(x_shape, nr_classes, n_models,
+                                              model_types=['CNN', "wrong_entry"])
+                                                                
```

Comment:
Nice test!
Note that this test should change a bit if you do what we suggested above, by specifying the error text.

Reply:
Done